### ISSUE TITLE
Got negative prediction value

### ISSUE THREAD

**Issue by Younger330** [bowang-lab/U-Mamba#63]
Hello, I used your model for training in our Trainer, but the prediction is negative
Here is my code. Am I wrong? Or would you have any suggestions?
`images=images.unsqueeze(4)
        out=umamba.forward(images)
       
        pred=out[0].squeeze(2)
        pred=F.log_softmax(pred,dim=1)
        #the predict is negative
`

**Comment by Younger330** [comment]
The problem is got by ` model.apply(InitWeights_He(1e-2)) `, but I dont know how it influence, maybe only used for radiological images?

Still have this question, although use the loss in nnunet

**Comment by Younger330** [comment]
> The problem is got by `model.apply(InitWeights_He(1e-2))`, but I dont know how it influence, maybe only used for radiological images?

Still have this problem after some iterations.

**Comment by Younger330** [comment]
Here is my training code:


    
    
    train_dataloader = DataLoader(train_dataset, batch_size=4, shuffle=True)
    
    
    
    umamba=get_umamba_bot_2d_from_plans(num_input_channels=3)
    umamba=umamba.cuda()
    
    
    epochs = 500
    umamba.train()
    optimizer = torch.optim.Adam(umamba.parameters(), lr=0.001)#,weight_decay=0.005)
    
    dc_cd_loss=DC_and_CE_loss()
    
    for epoch in tqdm(range(epochs),desc='Epoch'):
    
        epoch_loss=0
        mean_cross=0
        mean_dice=0
    
        for images, gt in tqdm(train_dataloader):
            images=images.cuda()
            images=images
            gt=gt.cuda()
            images=images.unsqueeze(4)
            out=umamba.forward(images)
            pred=out[0]
            pred=F.log_softmax(pred,dim=1)#negative prediction wheter delete or not
            gt=gt.unsqueeze(4)
    
            
            loss=dc_cd_loss(pred,gt)
    
    
            
            loss.backward()
            optimizer.step()
